toolchain/check/eval.cpp

```diff
@@ -17,6 +17,104 @@
 
 namespace Carbon::Check {
 
+namespace {
+// Information about an eval block of a specific that we are currently building.
+struct SpecificEvalInfo {
+  // The region within the specific whose eval block we are building.
+  SemIR::GenericInstIndex::Region region;
+  // The work-in-progress contents of the eval block.
+  llvm::ArrayRef<SemIR::InstId> values;
+};
+
+// Information about the context within which we are performing evaluation.
+struct EvalContext {
+  auto GetInContext(SemIR::ConstantId const_id) -> SemIR::ConstantId {
```

Comment:
Comment? It's not clear to me how this differs from the others, and it looks like a public API (though, if this is class-ified, should it be private?)

Reply:
Added a comment.